### PR TITLE
Updated bounds

### PR DESCRIPTION
**Pull request by CSVdB**
Note: Two tests fail for this code on my machine (NixOS), in DiagnoseSpec.hs.
I checked and the same two tests fail on my machine on your master branch.
Do you want me to look into why they fail?

### CODE REVIEW

File: test/SuperUserSpark/Compiler/Gen.hs [hunk 1]
```diff
@@ -55,6 +51,6 @@ instance GenUnchecked CompileError where
             [ PreCompileErrors <$> genUnchecked
             , DuringCompilationError <$> genUnchecked
             ]
+    shrinkUnchecked _ = []
```
Comment (NorfairKing) [inline test/SuperUserSpark/Compiler/Gen.hs]: This seems very wrong. Why did you add this?

File: stack.yaml [hunk 2]
```diff
@@ -1,16 +1,8 @@
...
-- genvalidity-path-0.1.0.0
-- directory-1.3.0.2
+- HUnit-1.5.0.0
 - iostring-0.0.0.0
-- optparse-applicative-0.13.1.0
-- process-1.5.0.0
-- validity-0.3.2.0
-- validity-path-0.1.0.0
-resolver: lts-7.19
+- path-0.5.13
```
Comment (NorfairKing) [inline stack.yaml]: I'd prefer to use newer paths. `>= 0.6` if we're changing versions anyway.

File: stack.yaml [hunk 3]
```diff
@@ -1,16 +1,8 @@
...
+- QuickCheck-2.9.2
```
Comment (NorfairKing) [inline stack.yaml]: QuickCheck is in the LTS too.

File: stack.yaml [hunk 4]
```diff
@@ -1,16 +1,8 @@
...
 packages:
 - '.'
 extra-deps:
-- aeson-1.1.0.0
-- genvalidity-0.3.1.0
-- genvalidity-hspec-0.3.1.0
-- genvalidity-hspec-aeson-0.0.1.0
-- genvalidity-path-0.1.0.0
-- directory-1.3.0.2
+- HUnit-1.5.0.0
```
Comment (NorfairKing) [inline stack.yaml]: I'm not sure why we need HUnit anymore.

**Comment by NorfairKing** [comment]
> Do you want me to look into why they fail?

That would be amazing!

File: stack.yaml (hunk 3 above)
Comment (Nickske666) [reply stack.yaml]: Sus depends on iostring, of which only one version exists (iostring-0.0.0.0),
and iostring depends on path =0.5.*. I can't update the path version to anything more recent
unless I change the bounds of iostring as well.

File: stack.yaml (hunk 4 above)
Comment (Nickske666) [reply stack.yaml]: Deleted

File: test/SuperUserSpark/Compiler/Gen.hs (hunk 1 above)
Comment (Nickske666) [reply test/SuperUserSpark/Compiler/Gen.hs]: Text.Parsec.Error.ParseError has no GenUnchecked and GenValid instances. Therefore, you can't derive these instances for ParseError (which you indeed didn't do, not sure whether you want to do this). Since ParseError is used in CompileError and CompileState, they can't derive shrinkUnchecked from Generic. This is why I added this line.
Note: The version of genvalidity which was used before I upgraded the bounds, didn't include any shrinking function yet. This is why I only needed to add this line now.
Possible solutions:
1. define this default shrinkUnchecked
2. define a GenValid instance for Text.Parsec.Error.ParseError, for your wrapper around that type, and derive shrinkUnchecked for CompileError and CompileState.
Which one do you prefer?

File: stack.yaml (hunk 2 above)
Comment (Nickske666) [reply stack.yaml]: The path bound can't be updated as long as the iostring bounds aren't updated.

File: stack.yaml (hunk 3 above)
Comment (NorfairKing) [reply stack.yaml]: We better get rid of iostring.
It's a fairly simple change to make.

File: test/SuperUserSpark/Compiler/Gen.hs (hunk 1 above)
Comment (NorfairKing) [reply test/SuperUserSpark/Compiler/Gen.hs]: Just leave this then, it's fine.
I forgot that shrinkUnchecked is new.

File: stack.yaml (hunk 2 above)
Comment (NorfairKing) [reply stack.yaml]: Get rid of iostring then. It's deprecated anyway.

File: super-user-spark.cabal [hunk 5]
```diff
@@ -54,23 +54,22 @@ library
...
+      , directory
+      , filepath
+      , mtl
+      , optparse-applicative
+      , parsec
+      , process
+      , hashable
+      , text
+      , unix
+      , validity              >= 0.4.0.4
```
Comment (NorfairKing) [inline super-user-spark.cabal]: make sure to upgrade the lowerbound on this if you have the new `path` version.

File: test/TestUtils.hs [hunk 6]
```diff
@@ -30,4 +30,4 @@ forFileInDirss dirs func =
             forM_ files func
 
 pend :: SpecWith ()
-pend = it "is still missing some tests" pending
+pend = it "is still missing some tests" pending
```
Comment (NorfairKing) [inline test/TestUtils.hs]: don't remove newlines plz

File: stack.yaml [hunk 7]
```diff
@@ -1,16 +1,12 @@
...
-- genvalidity-hspec-aeson-0.0.1.0
-- genvalidity-path-0.1.0.0
-- directory-1.3.0.2
-- iostring-0.0.0.0
-- optparse-applicative-0.13.1.0
-- process-1.5.0.0
-- validity-0.3.2.0
-- validity-path-0.1.0.0
-resolver: lts-7.19
+- QuickCheck-2.9.2
```
Comment (NorfairKing) [inline stack.yaml]: Again, why a custom(very old) version of quickcheck?

File: src/SuperUserSpark/Utils.hs [hunk 8]
```diff
@@ -42,4 +41,4 @@ containsMultipleConsequtiveSlashes = isInfixOf "//"
 (<**>) fa fb = do
     e1 <- fa
     e2 <- fb
-    pure $ e1 <*> e2
+    pure $ e1 <*> e2
```
Comment (NorfairKing) [inline src/SuperUserSpark/Utils.hs]: Please don't remove newlines.

File: stack.yaml [hunk 9]
```diff
@@ -1,15 +1,12 @@
...
-- genvalidity-hspec-0.3.1.0
-- genvalidity-hspec-aeson-0.0.1.0
-- genvalidity-path-0.1.0.0
-- directory-1.3.0.2
-- iostring-0.0.0.0
-- optparse-applicative-0.13.1.0
-- process-1.5.0.0
-- validity-0.3.2.0
-- validity-path-0.1.0.0
+- QuickCheck-2.9.2
```
Comment (NorfairKing) [inline stack.yaml]: This shouldn't be necessary

File: test/TestUtils.hs [hunk 10]
```diff
@@ -31,3 +31,4 @@ forFileInDirss dirs func =
 
 pend :: SpecWith ()
 pend = it "is still missing some tests" pending
+
```
Comment (NorfairKing) [inline test/TestUtils.hs]: Please don't add unnecessary newlines.

File: src/SuperUserSpark/Diagnose.hs [hunk 11]
```diff
@@ -69,12 +70,12 @@ diagnoseByCardRef checkCardReference = do
...
+    liftIO . LB.putStrLn $ JSON.encodePretty ddeps
 
 diagnoserBake :: SparkBaker a -> SparkDiagnoser a
 diagnoserBake =
     withExceptT DiagnoseBakeError .
     mapExceptT (withReaderT diagnoseBakeSettings)
 
 diagnoseDeployments :: [BakedDeployment] -> IO [DiagnosedDeployment]
-diagnoseDeployments = mapM diagnoseDeployment
+diagnoseDeployments = mapM diagnoseDeployment
```
Comment (NorfairKing) [inline src/SuperUserSpark/Diagnose.hs]: Again: why would you remove newlines.

File: test/SuperUserSpark/BakeSpec.hs [hunk 12]
```diff
@@ -169,4 +177,4 @@ bakeSpec =
                                 replaceId
                                     (env1 ++ [(var, val)] ++ env2)
                                     (Var var) `shouldBe`
-                                Right val
+                                Right val
```
Comment (NorfairKing) [inline test/SuperUserSpark/BakeSpec.hs]: Again, don't delete newlines please.

File: test/SuperUserSpark/DeployerSpec.hs [hunk 13]
```diff
@@ -254,4 +254,4 @@ deploymentSpec = do
                         removeDirRecur src
                         removeLink $ dropTrailingPathSeparator $ toFilePath dst
                         diagnoseFp src' `shouldReturn` Nonexistent
-                        diagnoseFp dst' `shouldReturn` Nonexistent
+                        diagnoseFp dst' `shouldReturn` Nonexistent
```
Comment (NorfairKing) [inline test/SuperUserSpark/DeployerSpec.hs]: newline.

File: src/SuperUserSpark/Bake.hs [hunk 14]
```diff
@@ -87,12 +86,12 @@ bakeByCardRef :: BakeCardReference -> SparkBaker ()
...
+    liftIO . LB.putStrLn $ JSON.encodePretty bdeps
 
 compileBakeCardRef :: BakeCardReference -> SparkBaker [RawDeployment]
 compileBakeCardRef (BakeCardCompiled fp) = bakerCompile $ inputCompiled fp
 compileBakeCardRef (BakeCardUncompiled bcf) = bakerCompile $ compileJob bcf
 
 bakerCompile :: ImpureCompiler a -> SparkBaker a
 bakerCompile =
-    withExceptT BakeCompileError . mapExceptT (withReaderT bakeCompileSettings)
+    withExceptT BakeCompileError . mapExceptT (withReaderT bakeCompileSettings)
```
Comment (NorfairKing) [inline src/SuperUserSpark/Bake.hs]: newline.

**Comment by CSVdB** [comment]
Are we still waiting for something to merge?
